src/Router.js

```diff
@@ -160,6 +161,10 @@ export default class Router {
 
       this.__currentCanonicalPath = canonicalPath
 
+      if (this.onRouteStart && mode !== 'replace') {
+        this.onRouteStart();
```

Comment:
Current canonical path might also be helpful for generalized onRouteStart functions. Would `ctx` be reasonable to be passed in as well?